[Bug]: test_bucket_notifs_shared_topic , test_multiple_bucket_notifs_setups, test_multi_notif_event_types - fail with similar error

https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/852/34376/1672187/1672247/log

https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/852/34376/1672187/1672246/log

https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/852/34376/1672187/1672245/log

In order to fix this issue we need a provider mode BM cluster. 
However, the fix should be also tested on a regular cluster. 

error : 

self = <mcg.test_bucket_notifications.TestBucketNotifications object at 0x7fb142afae50>
request = <SubRequest 'notif_manager' for <Function test_bucket_notifications[provided-logs-pvc]>>
pvc_factory_class = <function pvc_factory_fixture..factory at 0x7fb13a166040>


@pytest.fixture(autouse=True, scope="class")
def notif_manager(self, request, pvc_factory_class):
    """
    Set up Kafka and the BucketNotificationsManager

    Note that the dependency on the pvc_factory_class fixture is necessary
    to guarantee the correct teardown order. Otherwise the pvc factory teardown
    might fail when deleting a PVC that the BucketNotificationsManager is still using.

    Returns:
        BucketNotificationsManager: An instance of the BucketNotificationsManager class
    """
    notif_manager = BucketNotificationsManager()
    notif_manager.pvc_factory = pvc_factory_class
    request.addfinalizer(notif_manager.cleanup)

  notif_manager.setup_kafka()

tests/functional/object/mcg/test_bucket_notifications.py:68: 



ocs_ci/ocs/resources/bucket_notifications_manager.py:62: in setup_kafka
    self.amq.setup_amq_cluster(sc.name)
ocs_ci/ocs/amq.py:931: in setup_amq_cluster
    self.setup_amq_kafka_persistent(sc_name, size, replicas)
ocs_ci/ocs/amq.py:268: in setup_amq_kafka_persistent
    if self.is_amq_pod_running(
ocs_ci/ocs/amq.py:200: in is_amq_pod_running
    for pod in TimeoutSampler(



self = <ocs_ci.utility.utils.TimeoutSampler object at 0x7fb1408429a0>


def __iter__(self):
    if self.start_time is None:
        self.start_time = time.time()
    while True:
        self.last_sample_time = time.time()
        if self.timeout <= (self.last_sample_time - self.start_time):

          raise self.timeout_exc_cls(*self.timeout_exc_args)
E               ocs_ci.ocs.exceptions.TimeoutExpiredError: Timed out after 300s running get_pod_name_by_pattern("my-cluster", "myproject")



ocs_ci/utility/utils.py:1516: TimeoutExpiredError

--- a/ocs_ci/ocs/resources/bucket_notifications_manager.py
+++ b/ocs_ci/ocs/resources/bucket_notifications_manager.py
@@ -401,6 +401,10 @@ class BucketNotificationsManager:
                     event_dict = parsed_event["Records"][0]
                     events.append(event_dict)
         logger.info(events)
+
+        # Filter out the irrelevant TestEvent which helps NooBaa check the connection
+        events = [event for event in events if "TestEvent" not in str(event)]
+
         return events
 
     def cleanup(self):

--- a/tests/functional/object/mcg/test_bucket_notifications.py
+++ b/tests/functional/object/mcg/test_bucket_notifications.py
@@ -145,9 +145,7 @@ class TestBucketNotifications(MCGTest):
                 func=notif_manager.get_events,
                 topic=topic,
             ):
-                keys_in_notifs = set(
-                    event["s3"]["object"]["key"] for event in events[1:]
-                )
+                keys_in_notifs = set(event["s3"]["object"]["key"] for event in events)
                 delta = obj_keys_set.difference(keys_in_notifs)
                 if not delta:
                     logger.info("All expected events were received by Kafka")

